pkg/venafi/cloud/connector.go

```diff
@@ -135,35 +134,45 @@ func (c *Connector) Ping() (err error) {
 	return nil
 }
 
-// Authenticate authenticates the user with Venafi Cloud using the provided API Key
+// Authenticate sets the authentication credentials for the Venafi Cloud API.
+// It will send a request to the API to verify the credentials are correct.
 func (c *Connector) Authenticate(auth *endpoint.Authentication) error {
+	if err := c.SetAuthentication(auth); err != nil {
+		return err
+	}
+
+	if _, err := c.getUserDetails(); err != nil {
+		return fmt.Errorf("%w: %s", verror.AuthError, err)
+	}
+
+	return nil
+}
+
+// SetAuthentication sets the authentication credentials for the Venafi Cloud API.
+func (c *Connector) SetAuthentication(auth *endpoint.Authentication) (err error) {
+	defer func() {
+		if err != nil {
+			err = fmt.Errorf("%w: %s", verror.AuthError, err)
+		}
+	}()
+
 	if auth == nil {
 		return fmt.Errorf("failed to authenticate: missing credentials")
 	}
 
-	//1. Access token. Assign it to connector
 	if auth.AccessToken != "" {
+		// 1. Access token. Assign it to connector
 		c.accessToken = auth.AccessToken
 	} else if auth.TokenURL != "" && auth.ExternalJWT != "" {
-		//2. JWT and token URL. use it to request new access token
+		// 2. JWT and token URL. use it to request new access token
 		tokenResponse, err := c.GetAccessToken(auth)
 		if err != nil {
 			return err
 		}
 		c.accessToken = tokenResponse.AccessToken
 	} else if auth.APIKey != "" {
-		// 3. API key. Get user to test authentication
+		// 3. API key. Assign it to connector
 		c.apiKey = auth.APIKey
-		url := c.getURL(urlResourceUserAccounts)
```

Comment:
This check has been moved to the `getUserDetails` function which is called by `Authenticate`.